### PR TITLE
Add zonal stats and implement count for Statistics methods

### PR DESCRIPTION
**Pull request by moradology**
1. Implement a zonalStatistics method
2. Remove no-op loop inside the statistics generating code
3. Add a data count on the Statistics object

### CODE REVIEW

File: raster-test/src/test/scala/geotrellis/raster/op/stats/StatsMethodsSpec.scala [hunk 1]
```diff
@@ -39,7 +39,7 @@ class StatsMethodsSpec extends FunSpec
       val stats = RasterSource.fromPath("raster-test/data/quad8.arg").get.statistics
 
       val dev = math.sqrt((2 * (0.5 * 0.5) + 2 * (1.5 * 1.5)) / 4)
-      val expected = Statistics(2.5, 3, 1, dev, 1, 4)
+      val expected = Statistics(400, 2.5, 3, 1, dev, 1, 4)
```
Comment (lossyrob) [inline raster-test/src/test/scala/geotrellis/raster/op/stats/StatsMethodsSpec.scala]: What happened here? Was a test passing when it shouldn't have been?

File: raster-test/src/test/scala/geotrellis/raster/op/zonal/ZonalStatisticsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+import geotrellis.raster.op.stats._
+
+import geotrellis.raster._
+
+import org.scalatest._
+
+import geotrellis.testkit._
+
+import scala.collection.mutable
+
```
Comment (lossyrob) [inline raster-test/src/test/scala/geotrellis/raster/op/zonal/ZonalStatisticsSpec.scala]: Shouldn't space out all the imports like this

File: raster/src/main/scala/geotrellis/raster/histogram/Histogram.scala [hunk 3]
```diff
@@ -136,12 +134,12 @@ abstract trait Histogram extends Serializable {
       var needMedian = true
       val limit = getTotalCount() / 2
 
-      var i = 0
       val len = values.length
 
-      while (i < len) {
+      cfor(0)(_ < len, _ + 1) { i =>
```
Comment (lossyrob) [inline raster/src/main/scala/geotrellis/raster/histogram/Histogram.scala]: Sometimes a while loop is faster than a cfor. Don't need to change these. We'll keep these in, but then benchmark the old vs new histogram and maybe change back, just wanted to give a heads up

Comment (moradology) [inline raster/src/main/scala/geotrellis/raster/histogram/Histogram.scala]: Good to know. I had looked into it and figured out that cfor was basically just being compiled into a while loop but figured that the consistency may be worth it. Will bench this and post something on this PR.

**Pull request merged**